provider/core.rb

```diff
@@ -602,8 +602,18 @@ def generate_file(data)
     end
 
     def generate_file_write(ctx, data)
-      Google::LOGGER.info "Generating #{data[:name]} #{data[:type]}"
-      write_file data[:out_file], compile_file(ctx, data[:template])
+      enforce_file_expectations data[:out_file] do
+        Google::LOGGER.info "Generating #{data[:name]} #{data[:type]}"
+        write_file data[:out_file], compile_file(ctx, data[:template])
+      end
+    end
+
+    def enforce_file_expectations(filename)
```

Comment:
The idea is that we could add more enforcement mechanisms like this as time goes by...

(We can't do that for copyright because the copyright header copyrights both the generated code and the template at the same time)